ext/openssl/ossl_x509cert.c

```diff
@@ -704,6 +704,94 @@ ossl_x509_eq(VALUE self, VALUE other)
     return !X509_cmp(a, b) ? Qtrue : Qfalse;
 }
 
+static VALUE
+ossl_x509_load_chained_cert_from_file(VALUE self, VALUE path)
+{
+    VALUE certs; /* Return array (Certificate object) */
+    BIO *in;
+    X509 *x509, *ca;
+    STACK_OF(X509) *certs_stack;
+    unsigned long err;
+    SSL_CTX *ctx = SSL_CTX_new(TLS_method());
```

Comment:
You should be able to avoid the need for this, and just create an array and put the certificates into that array below: